### DISCUSSION TITLE
Difficulty implementing ORDER BY with a findOne

### DISCUSSION THREAD

**Discussion by wolfderby** [sequelize/sequelize#15548] [Help & Questions]
I'm new to sequelize and nodejs, so hopefully this is something obvious

when I implement this `order` for an ORDER BY, I get this error:

`SequelizeDatabaseError: column NaturalProduct.total does not exist`

I've been all over the docs and google for the last 2 days and nothing seems to be working

please help! 

model:
```
        NaturalProduct.hasMany(PlantOutcome, {
            as: 'plantOutcomes',
            foreignKey: {
                name: 'naturalProductUid',
                field: 'natural_product_uid',
                allowNull: true
            },
            timestamps: false
        });
```
controller:
```
            NaturalProduct.findOne({
                    where: {'uid': id},
                    include: [
                        {
                            model: Concept,
                            as: 'concept'
                        },
                        {
                            model: PlantCommonName,
                            as: 'plantCommonNames'
                        },
                        {
                            model: PlantOutcome,
                            as: 'plantOutcomes',
                            // include:[{
                            //     model: aPlantOutcome,
                            //     as: aPlantOutcome,
                            // }]
                        }
                    ],
                    order: [
                                  'total', 'DESC'                                                                                                     
                     ]
                     })];
```

**Comment by joseneves-dev** [comment]
There is several issues there. 
to do a order you have to do something like this 

`NaturalProduct.findOne({
                    where: {'uid': id},
                    include: [
                        {
                            model: Concept,
                            as: 'concept'
                        },
                        {
                            model: PlantCommonName,
                            as: 'plantCommonNames'
                        },
                        {  model: PlantOutcome,
                            as: 'plantOutcomes',
                            // include:[{
                            //     model: aPlantOutcome,
                            //     as: aPlantOutcome,
                            // }]
                        }
                    ],
                    order: [[ 'model_name', 'total', 'DESC' ]]
          });`
                   
Because i assume that model NaturalProduct dosen't have a field called total, so you have to say what model you want to order by

**Reply by fathe4** [reply]
HI @FreeLancerPT, can you help me with this question, if you can? Thank you

**Reply by joseneves-dev** [reply]
What question?  @fathe be more specific please
